xdsl/transforms/constant_fold_interp.py

```diff
@@ -27,7 +27,7 @@ class ConstantFoldInterpPattern(RewritePattern):
     interpreter: Interpreter
 
     def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
-        if not op.has_trait(Pure):
```

Comment:
I'd prefer either having Constantlike inherit from Pure or doing this in a different PR.

Reply:
I agree, could you please split this out into a separate PR?